File: react-type-vite/src/services/api/response/userResponse.ts
import type { AccountStatus } from "@/types/account.enum";

interface UserResponse {
  id: string;
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  avatarUrl?: string;
  dob: string;
  roles: string[];
  accountStatus: AccountStatus;
}

export type { UserResponse };

File: spring-microservices/identity-service/src/main/java/com/devteria/identity/dto/response/UserResponse.java
package com.devteria.identity.dto.response;

import java.time.LocalDate;
import java.util.Set;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UserResponse {
    String id;
    String username;
    String email;
    String phoneNumber;
    String firstName;
    String lastName;
    String avatarUrl;
    String accountStatus;
    LocalDate dob;
    String bio;
    //    Set<String> favoriteCategories;
    String role;
    String createdAt;
}
